ui/packages/catalog-ui-search/src/main/webapp/js/widgets/openlayers.polygon.js

```diff
@@ -11,252 +11,252 @@
+        }
+
+        const bufferWidth = DistanceUtils.getDistanceInMeters(this.model.get('polygonBufferWidth'), this.model.get('polygonBufferUnits')) || 1;
+        const line = Turf.lineString(translateFromOpenlayersCoordinates(rectangle.getCoordinates()));
+        const bufferedLine = Turf.buffer(line, bufferWidth, 'meters');
+        const geometryRepresentation = (bufferedLine && new ol.geom.Polygon(bufferedLine.geometry.coordinates)) || rectangle;
+
+        this.billboard = new ol.Feature({
+            geometry: geometryRepresentation
+        });
 
-                var color = this.model.get('color');
+        const color = this.model.get('color');
 
-                var iconStyle = new ol.style.Style({
-                    stroke: new ol.style.Stroke({ color: color ? color : '#914500', width: 3 })
-                });
-                this.billboard.setStyle(iconStyle);
+        const iconStyle = new ol.style.Style({
+            stroke: new ol.style.Stroke({ color: color ? color : '#914500', width: 3 })
+        });
+        this.billboard.setStyle(iconStyle);
 
-                var vectorSource = new ol.source.Vector({
-                    features: [this.billboard]
-                });
+        const vectorSource = new ol.source.Vector({
+            features: [this.billboard]
+        });
 
-                var vectorLayer = new ol.layer.Vector({
-                    source: vectorSource
-                });
+        const vectorLayer = new ol.layer.Vector({
+            source: vectorSource
+        });
 
-                this.vectorLayer = vectorLayer;
-                this.map.addLayer(vectorLayer);
-            },
-
-            handleRegionStop: function(sketchFeature) {
-                const geometry = olUtils.wrapCoordinatesFromGeometry(sketchFeature.feature.getGeometry());
-                this.setModelFromGeometry(geometry);
-                this.drawBorderedPolygon(geometry);
-                this.listenTo(this.model, 'change:polygon', this.updateGeometry);
-
-                this.model.trigger("EndExtent", this.model);
-                wreqr.vent.trigger('search:polydisplay', this.model);
-            },
-            start: function() {
-                var that = this;
-
-                this.primitive = new ol.interaction.Draw({
-                    type: 'Polygon',
-                    style: new ol.style.Style({
-                        stroke: new ol.style.Stroke({
-                            color: [0, 0, 255, 0]
-                        })
-                    })
-                });
+        this.vectorLayer = vectorLayer;
+        this.map.addLayer(vectorLayer);
+    },
+
+    handleRegionStop: function(sketchFeature) {
+        const geometry = olUtils.wrapCoordinatesFromGeometry(sketchFeature.feature.getGeometry());
+        this.setModelFromGeometry(geometry);
+        this.drawBorderedPolygon(geometry);
+        this.listenTo(this.model, 'change:polygon', this.updateGeometry);
```

Comment:
✏️`this.listenTo(this.model, 'change:polygon change:polygonBufferWidth', this.updateGeometry);`